### ISSUE TITLE
Subtyping property

### ISSUE THREAD

**Issue by relsunkaev** [microsoft/pyright#2935]
**Describe the bug**
Pyright does not recognize attributes defined in addition to those defined in `property` when subtyping from it.

**To Reproduce**
The following code
```python
class custom_property(property):
    _custom_func: Callable[..., Any] | None
    def custom_function(self, _custom_func: Callable[..., Any]):
        self._custom_func = _custom_func

class SomeClass:
    _custom_value: int | None
    @custom_property
    def custom_value(self) -> int | None:
        return self._custom_value

    @some_func.setter
    def custom_value(self, value: int):
        self.cusom_value = value

    @some_func.custom_function
    def custom_value(self):
        pass
```
will produce the following errors
```
test.py(11, 9): Method declaration "custom_value" is obscured by a declaration of the same name.
test.py(15, 9): Method declaration "custom_value" is obscured by a declaration of the same name.
test.py(18, 19): Cannot access member "custom_function" for type "custom_property". Member "custom_function" is unknown.
```
However, if the
```python
@some_func.custom_function
def custom_value(self):
    pass
```
declaration is removed from `SomeClass`, no errors are produced.

**Expected behavior**
No errors.

**Versions**
VS Code: 1.63.2
Pylance: v2022.1.4-pre.1

**Comment by erictraut** [comment]
Thanks for the bug report. 

Pyright contains a bunch of special-case logic for the built-in `property` class. This logic didn't handle subclasses of `property` very well. I've improved the code so it avoids the three errors that your sample code previously produced.

I needed to make a few changes to your sample. I think these were just typos, but I wanted to mention this for completeness. Here's the modified version:

```python
class custom_property(property):
    _custom_func: Callable[..., Any] | None
    def custom_function(self, _custom_func: Callable[..., Any]):
        self._custom_func = _custom_func
        return self

class SomeClass:
    _custom_value: int | None
    @custom_property
    def custom_value(self) -> int | None:
        return self._custom_value

    @custom_value.setter
    def custom_value(self, value: int):
        self._custom_value = value

    @custom_value.custom_function
    def custom_value(self):
        pass
```

**Comment by relsunkaev** [comment]
Awesome! And yeah, those were just typos. `custom_value` was called `some_func` and I forgot to edit the decorators (along with the the `return self`).

**Comment by erictraut** [comment]
This is included in pyright 1.1.215, which I just published. It will also be included in the next release of pylance.
